src/Kernel/Context.class.st

```diff
@@ -292,11 +292,7 @@ Context >> activeOuterContext [
 Context >> arguments [
 	"returns the arguments of a message invocation"
 	
-	| arguments numargs |
-	numargs :=  self compiledCode numArgs.
-	arguments := Array new: numargs.
-	1 to: numargs do: [ :i | arguments at: i put: (self tempAt: i) ].
-	^ arguments
+	^(1 to: self numArgs) collect: [:i | self tempAt: i]
```

Comment:
I believe that `to:do:` is inlined? Do we feel the loss of this optimization here?
But maybe, since we are dealing with Context, and possibly metaprograming or debuging (who else cares about a collection of the arguments of a method?), the loss is probably negligible.

Reply:
Yes, it is less optimized (it creates the intermediate interval, uses a real closure). But it is not used anywhere right now  (the debugger does not call it to show the stack)